### ISSUE TITLE
Update Azure integration logging functionality

### ISSUE THREAD

**Issue by CarlosRS9** [wazuh/wazuh#11938]
| Affected integration |
|---|
| Azure |

### Azure-logs.log file
Our integration with Azure currently displays relevant information about each execution (Info, Debug, Warning and Error messages) in a custom log file called `azure-logs.log`, instead of being consistent with the AWS and GCP modules which generate the same type of messages at runtime but logs them into the `ossec.log` file instead.

We must update the Azure module to stop using this custom log file in favor of ossec.log.

The following is an example of the messages generated during an execution of the Azure module, to demonstrate that including these messages in the ossec.log would not generate any problems since the number of messages is relatively low:

```
01/21/2022 03:28:37 PM INFO: AZURE Getting the data from /var/ossec/wodles/azure/last_dates.json.
01/21/2022 03:28:37 PM INFO: AZURE Azure Storage starting.
01/21/2022 03:28:37 PM INFO: AZURE Storage: Authenticating.
01/21/2022 03:28:37 PM INFO: AZURE Storage: Authenticated.
01/21/2022 03:28:37 PM INFO: AZURE 82e049b81fa6eb88ebf85f1677785f2b was not found in /var/ossec/wodles/azure/last_dates.json for storage
01/21/2022 03:28:37 PM INFO: AZURE Storage: Getting blobs.
01/21/2022 03:28:38 PM INFO: AZURE Storage: The search starts from the date: 2021-12-02 15:28:37.934631+00:00 for blobs in container: 'frameworktestcontainer' 
01/21/2022 03:28:38 PM INFO: AZURE Updating /var/ossec/wodles/azure/last_dates.json file.
01/21/2022 03:28:38 PM INFO: AZURE Storage: End
```
### Modules debug level

Another feature the Azure integration is missing and differentiates it from other integrations such as the AWS one is the ability to choose different levels of verbosity depending on the value the user has assigned to the `wazuh_modules.debug` in the `local_internal_options.conf` or the `internal_options.conf` configuration files. 

While the AWS integration will show more or less messages during its execution depending on the `debug` parameter, the Azure integration only has the ability to enable or disable debug messages, but does not allow the user to choose the desired verbosity.

We should update this to work like the AWS integration.

This change requires to update both the Core and Python parts of the implementation.

## Tasks
- [x] Deprecate azure-logs.log file and replace it with ossec.log.
- [x] Update debug parameter in Core parser.
- [x] Update debug parameter in the Python module.
- [x] Update Core Azure Unit Tests.
- [x] Test in a manager.
- [x] Test in an agent.
- [x] Unit tests without failures. Updated if there are any relevant changes.
- [x] integration tests without failures. Updated if there are any relevant changes.
- [x] Update the documentation if necessary.
- [x] Add entry to the changelog if necessary.

**Comment by CarlosRS9** [comment]
# Status Update

We have applied the necessary changes so that the Azure module displays its logs in the `ossec.log` file instead of its own log file. Because the gcloud module is not consistent in its use of the debug parameter we have not been able to move this logic to the `utils.py` file common to all modules yet. This change will be applied once the #10334 is finished.

The Core part is still pending. We are currently working on this.

**Comment by CarlosRS9** [comment]
# Status update

We have finished applying all the necessary changes, both in the python andthe core parts.


The most important changes are detailed below:


- The message management is now handled by the `logging` library, in a similar way to how the `gcloud` module works. The `logging` module has been configured to send the different debug messages to `stdout`, since the core part is designed to capture this `stdout` and show it in the `ossec.log`. The rest of the modules are also implemented this way, so it is consistent.

- The `-v` or `--verbosity` parameter has been deprecated. This parameter was used to set the module to send the different debug messages to the `stdout`. When it was not present, they were sent to the `azure-logs.log` file instead. This parameter is no longer needed. It has been kept in the module to mantain compatibility and to show a warning to the user when trying to use it.

- A new debug level parameter called `--debug` or `-d` has been added. It works ìn a similar way to the other modules. The higher the value indicated, the higher the verbosity, being the accepted values 0, 1 or 2. 

- The Core code has been updated to make use of the new debug parameter.

- The Core Unit Tests have been revised to ensure that no compatibility is broken after the changes.



## Testing

For testing purposes, the following configuration has been executed. It was designed to fail, since it does not indicate any of the mandatory parameters for authentication:

<details>
  <summary>Azure configuration</summary>

	<wodle name="azure-logs">
		<disabled>no</disabled>
		<interval>10m</interval>
		<run_on_start>yes</run_on_start>

		<storage>
			<tag>storage</tag>
			<auth_path>invalid</auth_path>
			<container name="invalid">
				<time_offset>50d</time_offset>
			</container>
		</storage>

		<log_analytics>
			<auth_path>invalid</auth_path>
			<tenantdomain>invalid</tenantdomain>
			<request>
				<tag>log-analytics</tag>
				<query>AzureDiagnostics</query>
				<workspace>invalid</workspace>
			</request>
		</log_analytics>

	    <graph>
			<auth_path>invalid</auth_path>
			<tenantdomain>invalid</tenantdomain>
	        <request>
	        	<tag>graph</tag>
	            <query>invalid</query>
	            <time_offset>1d</time_offset>
	        </request>
	    </graph>
	</wodle>
</details>


Here is the output of the module when using the different modules debug modes available:


<details>
  <summary>modules.debug=0</summary>

	2022/02/24 14:54:04 wazuh-modulesd:azure-logs: INFO: Module started.
	2022/02/24 14:54:04 wazuh-modulesd:azure-logs: INFO: Starting fetching of logs.
	2022/02/24 14:54:04 wazuh-modulesd:azure-logs: INFO: Starting Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: ERROR: log-analytics: Returned error code: '1'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Log Analytics collection for request 'log-analytics'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Starting Graphs log collection for the domain 'invalid'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: ERROR: graph: Returned error code: '1'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for request 'graph'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Graphs log collection for the domain 'invalid'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Starting Storage log collection for 'storage'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: ERROR: invalid: Returned error code: '1'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for container 'invalid'.
	2022/02/24 14:54:05 wazuh-modulesd:azure-logs: INFO: Finished Storage log collection for 'storage'.
</details>

<details>
  <summary>modules.debug=1</summary>

	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:53 at wm_azure_main(): INFO: Module started.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:73 at wm_azure_main(): INFO: Starting fetching of logs.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:80 at wm_azure_main(): INFO: Starting Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:171 at wm_azure_log_analytics(): DEBUG: Launching command: wodles/azure/azure-logs --log_analytics --la_auth_path invalid --la_tenant_domain invalid --la_tag log-analytics --la_query "AzureDiagnostics" --workspace invalid --debug 1
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:175 at wm_azure_log_analytics(): ERROR: log-analytics: Returned error code: '1'.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:176 at wm_azure_log_analytics(): DEBUG: OUTPUT: 2022/02/24 02:53:12 azure: INFO: Sample message
	2022/02/24 02:53:12 azure: WARNING: Sample message
	2022/02/24 02:53:12 azure: ERROR: Sample message
	2022/02/24 02:53:12 azure: CRITICAL: Sample message
	2022/02/24 02:53:12 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:53:12 azure: INFO: Azure Log Analytics starting.
	2022/02/24 02:53:12 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:188 at wm_azure_log_analytics(): INFO: Finished Log Analytics collection for request 'log-analytics'.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:82 at wm_azure_main(): INFO: Finished Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:84 at wm_azure_main(): INFO: Starting Graphs log collection for the domain 'invalid'.
	2022/02/24 14:53:12 wazuh-modulesd:azure-logs[30520] wm_azure.c:258 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path invalid --graph_tenant_domain invalid --graph_tag graph --graph_query 'invalid' --graph_time_offset 1d --debug 1
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:262 at wm_azure_graphs(): ERROR: graph: Returned error code: '1'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:263 at wm_azure_graphs(): DEBUG: OUTPUT: 2022/02/24 02:53:13 azure: INFO: Sample message
	2022/02/24 02:53:13 azure: WARNING: Sample message
	2022/02/24 02:53:13 azure: ERROR: Sample message
	2022/02/24 02:53:13 azure: CRITICAL: Sample message
	2022/02/24 02:53:13 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:53:13 azure: INFO: Azure Graph starting.
	2022/02/24 02:53:13 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:275 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'graph'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:86 at wm_azure_main(): INFO: Finished Graphs log collection for the domain 'invalid'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:91 at wm_azure_main(): INFO: Starting Storage log collection for 'storage'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:358 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path invalid --container "invalid" --blobs "*" --storage_tag storage --storage_time_offset 50d --debug 1
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:362 at wm_azure_storage(): ERROR: invalid: Returned error code: '1'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:363 at wm_azure_storage(): DEBUG: OUTPUT: 2022/02/24 02:53:13 azure: INFO: Sample message
	2022/02/24 02:53:13 azure: WARNING: Sample message
	2022/02/24 02:53:13 azure: ERROR: Sample message
	2022/02/24 02:53:13 azure: CRITICAL: Sample message
	2022/02/24 02:53:13 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:53:13 azure: INFO: Azure Storage starting.
	2022/02/24 02:53:13 azure: INFO: Storage: Authenticating.
	2022/02/24 02:53:13 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:375 at wm_azure_storage(): INFO: Finished Storage log collection for container 'invalid'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:93 at wm_azure_main(): INFO: Finished Storage log collection for 'storage'.
	2022/02/24 14:53:13 wazuh-modulesd:azure-logs[30520] wm_azure.c:99 at wm_azure_main(): DEBUG: Fetching logs finished.
</details>

<details>
  <summary>modules debug 2 - Output after the change</summary>

	2022/02/24 14:49:37 wazuh-modulesd[29994] main.c:94 at main(): DEBUG: Created new thread for the 'azure-logs' module.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:53 at wm_azure_main(): INFO: Module started.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:73 at wm_azure_main(): INFO: Starting fetching of logs.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:80 at wm_azure_main(): INFO: Starting Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:119 at wm_azure_log_analytics(): DEBUG: Creating argument list.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:171 at wm_azure_log_analytics(): DEBUG: Launching command: wodles/azure/azure-logs --log_analytics --la_auth_path invalid --la_tenant_domain invalid --la_tag log-analytics --la_query "AzureDiagnostics" --workspace invalid --debug 2
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:175 at wm_azure_log_analytics(): ERROR: log-analytics: Returned error code: '1'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:176 at wm_azure_log_analytics(): DEBUG: OUTPUT: 2022/02/24 02:49:37 azure: INFO: Sample message
	2022/02/24 02:49:37 azure: DEBUG: Sample message
	2022/02/24 02:49:37 azure: WARNING: Sample message
	2022/02/24 02:49:37 azure: ERROR: Sample message
	2022/02/24 02:49:37 azure: CRITICAL: Sample message
	2022/02/24 02:49:37 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:49:37 azure: INFO: Azure Log Analytics starting.
	2022/02/24 02:49:37 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:188 at wm_azure_log_analytics(): INFO: Finished Log Analytics collection for request 'log-analytics'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:82 at wm_azure_main(): INFO: Finished Log Analytics collection for the domain 'invalid'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:84 at wm_azure_main(): INFO: Starting Graphs log collection for the domain 'invalid'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:208 at wm_azure_graphs(): DEBUG: Creating argument list.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:258 at wm_azure_graphs(): DEBUG: Launching command: wodles/azure/azure-logs --graph --graph_auth_path invalid --graph_tenant_domain invalid --graph_tag graph --graph_query 'invalid' --graph_time_offset 1d --debug 2
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:262 at wm_azure_graphs(): ERROR: graph: Returned error code: '1'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:263 at wm_azure_graphs(): DEBUG: OUTPUT: 2022/02/24 02:49:37 azure: INFO: Sample message
	2022/02/24 02:49:37 azure: DEBUG: Sample message
	2022/02/24 02:49:37 azure: WARNING: Sample message
	2022/02/24 02:49:37 azure: ERROR: Sample message
	2022/02/24 02:49:37 azure: CRITICAL: Sample message
	2022/02/24 02:49:37 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:49:37 azure: INFO: Azure Graph starting.
	2022/02/24 02:49:37 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:275 at wm_azure_graphs(): INFO: Finished Graphs log collection for request 'graph'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:86 at wm_azure_main(): INFO: Finished Graphs log collection for the domain 'invalid'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:91 at wm_azure_main(): INFO: Starting Storage log collection for 'storage'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:296 at wm_azure_storage(): DEBUG: Creating argument list.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:358 at wm_azure_storage(): DEBUG: Launching command: wodles/azure/azure-logs --storage --storage_auth_path invalid --container "invalid" --blobs "*" --storage_tag storage --storage_time_offset 50d --debug 2
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:362 at wm_azure_storage(): ERROR: invalid: Returned error code: '1'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:363 at wm_azure_storage(): DEBUG: OUTPUT: 2022/02/24 02:49:37 azure: INFO: Sample message
	2022/02/24 02:49:37 azure: DEBUG: Sample message
	2022/02/24 02:49:37 azure: WARNING: Sample message
	2022/02/24 02:49:37 azure: ERROR: Sample message
	2022/02/24 02:49:37 azure: CRITICAL: Sample message
	2022/02/24 02:49:37 azure: INFO: Getting the data from /var/ossec/wodles/azure/last_dates.json.
	2022/02/24 02:49:37 azure: INFO: Azure Storage starting.
	2022/02/24 02:49:37 azure: INFO: Storage: Authenticating.
	2022/02/24 02:49:37 azure: ERROR: Error: The authentication file could not be opened: [Errno 2] No such file or directory: 'invalid'
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:375 at wm_azure_storage(): INFO: Finished Storage log collection for container 'invalid'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:93 at wm_azure_main(): INFO: Finished Storage log collection for 'storage'.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:99 at wm_azure_main(): DEBUG: Fetching logs finished.
	2022/02/24 14:49:37 wazuh-modulesd:azure-logs[29994] wm_azure.c:69 at wm_azure_main(): DEBUG: Sleeping until: 2022/02/24 14:59:37
</details>



As a reference, here is the output provided by the module when using the same configuration but without the changes introduced in this issue:

<details>
  <summary>modules.debug=0</summary>
</details>

<details>
  <summary>modules.debug=1</summary>
</details>

<details>
  <summary>modules.debug=2</summary>
</details>
